apis/python/src/tiledbsoma/factory.py

```diff
@@ -69,27 +69,27 @@ def _construct_member(
     assert class_name is not None
 
     # Now invoke the appropriate per-class constructor.
-    if class_name == "SOMAExperiment":
+    if class_name == "Experiment":
```

Comment:
A nit: we are changing the metadata value due to a change in the Python API, which implies they are linked somehow.  That seems a bit awkward as we may have other Python API-only changes in the future that don't change the persistent data storage format.

What would you think about creating a separate enumeration of values that are strored in the metadata, rather than storing the (literal) class name?   Eg, store 0 for Collection, 1 for DataFrame, etc.  Put another way, a level of indirection between the class name and the metadata code.

Eg, in Python:
```
class PersistentMetadataObjectTypes(Enum):
    SOMA_COLLECTION = 0,
    SOMA_DATAFRAME = 1,
    ...
```

In practice, I'm not sure this buys us more than a more explicit separation between the per-language class name and the underlying metadata.

Reply:
@bkmartinjr very good point.

my recommendation is that we use the classnames from the language-independent API spec. then in `factory.py` we can have a dict mapping between API-spec names like `SOMADataFrame` and language-specific names like `DataFrame`.

Reply:
i prefer this over int-codes as the int codes are really just a less-clear enumeration of the API-spec names; for clarity i'd prefer to store the latter as metadata

Reply:
Works for me.  My main point was not assume class_name == metadata_code.  Using the abstract type string works for me, which I take to mean `SOMACollection`, `SOMADataFrame`, etc.